casper/src/main/scala/coop/rchain/casper/TuplespaceEvent.scala

```diff
@@ -20,31 +20,38 @@ case object Receive extends Polarity
 trait Cardinality
 case object Linear    extends Cardinality
 case object NonLinear extends Cardinality
+case object Peek      extends Cardinality
 
 object TuplespaceEvent {
 
   private[this] def toOperation(produce: Produce): TuplespaceOperation =
     TuplespaceOperation(Send, if (produce.persistent) NonLinear else Linear, produce.hash)
 
-  private[this] def toOperation(consume: Consume): TuplespaceOperation =
-    TuplespaceOperation(Receive, if (consume.persistent) NonLinear else Linear, consume.hash)
+  private[this] def toOperation(consume: Consume, peek: Boolean): TuplespaceOperation =
+    TuplespaceOperation(
+      Receive,
+      if (consume.persistent) NonLinear else if (peek) Peek else Linear,
```

Comment:
my OCD says "this needs to be extracted to a `match`" but I guess it's fine. ifs are faster, right?

Reply:
Right :-)